alacritty_terminal/src/ansi.rs

```diff
@@ -1080,12 +1080,18 @@ where
             },
             ('r', None) => {
                 let arg0 = arg_or_default!(idx: 0, default: 1) as usize;
+                let arg1 = arg_or_default!(idx: 1, default: handler.lines().0 as _) as usize;
+
+                if arg0 >= arg1 {
```

Comment:
I'm just constantly follow old code in that hot fixes for whatever reason.

Reply:
I think rather than shadowing (rebinding) `top` and `bottom` we could just use call `set_scrolling_region(Line(top - 1)..Line(bottom))` here honestly.

This is a comment of arguable merit, please feel free to ignore.

Reply:
I think I prefer shadowing, especially because of the comment explaining that the subtraction here is intentional.